### PR TITLE
Add Regression and Classification Sklearn Metrics to Directly Use mlflow.metrics.make_metric

### PR DESCRIPTION
**Pull request by jessechancy**
## What changes are proposed in this pull request?

Add additional sklearn metrics to use make_metric. Also, a change to the function signature that includes additional parameters for metrics and a change from eval_df Dataframe to y_true and y_pred series.

## How is this patch tested?

- [ ] Existing unit/integration tests
- [x] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)
Refer to https://github.com/mlflow/mlflow/pull/9515

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/metrics/__init__.py [hunk 1]
```diff
@@ -191,6 +200,126 @@
 .. _rougeLsum: https://huggingface.co/spaces/evaluate-metric/rouge
 """
 
+# General Regression Metrics
+
+mae = make_metric(
+    eval_fn=_mae_eval_fn,
+    greater_is_better=False,
+    name="mae",
```
Comment (annzhang-db) [inline mlflow/metrics/__init__.py:208]: We don't want to change the names from how they are logged right now in mlflow.evaluate, look in `_get_regressor_metrics` in `default_evaluator.py` to see how they are named rn (e.x. mean_absolute_error for this specific one).

File: mlflow/metrics/utils/metric_definitions.py [hunk 2]
```diff
@@ -249,3 +249,102 @@ def _rougeLsum_eval_fn(eval_df, metrics):
...
+
+
+def _mae_eval_fn(eval_df, metrics):
+    if "target" in eval_df:
+        from sklearn.metrics import mean_absolute_error
+
+        y_pred = eval_df["prediction"]
+        y = eval_df["target"]
+
+        mae = mean_absolute_error(y, y_pred)
```
Comment (annzhang-db) [inline mlflow/metrics/utils/metric_definitions.py:261]: These are missing `sample_weight` compared to how they are [calculated now in mlflow.evaluate](https://github.com/mlflow/mlflow/blob/7a592885ca4c827e6951e06ebab5fcbd614e56e4/mlflow/models/evaluation/default_evaluator.py#L122) - do we want to add this as an input to eval_fn in make_metric? cc @sunishsheth2009 @prithvikannan

**Review by sunishsheth2009 — APPROVED** [review]

**Review by annzhang-db — CHANGES_REQUESTED** [review]
Added a few comments but looking good! :)

File: mlflow/metrics/utils/metric_definitions.py (hunk 2 above)
Comment (sunishsheth2009) [reply mlflow/metrics/utils/metric_definitions.py:261]: Ahh interesting, missed that. This brings us to the point of we need some things from the evaluator_configs to compute the results. Need to think about this on how we can get that information.
cc @dbczumar to get some thoughts here since we explicitly decide to now include evaluator_config in make_metric.

Comment (jessechancy) [reply mlflow/metrics/utils/metric_definitions.py:261]: In addition to `sample_weight`, there are also parameters for the classification metrics that help indicate whether they are for binary or multiclass classification. If we have a evaluator_config, we may also want to be able to pass in that information.

File: mlflow/metrics/__init__.py (hunk 1 above)
Comment (jessechancy) [reply mlflow/metrics/__init__.py:208]: okay I'll keep the full names

File: mlflow/metrics/utils/metric_definitions.py (hunk 2 above)
Comment (dbczumar) [reply mlflow/metrics/utils/metric_definitions.py:261]: @sunishsheth2009 @jessechancy I think `pos_label` and `sample_weights` should be optional variables defined on the metric. Metrics have the concept of variables, right?
